sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -495,8 +480,11 @@ object ScalaJSPluginInternal {
 
   val scalaJSTestFrameworkSettings = Seq(
       loadedTestFrameworks := {
-        // use assert to prevent warning about pure expr in stat pos
-        assert(scalaJSEnsureUnforked.value)
+        if (fork.value) {
+          throw new MessageOnlyException(
+              "`test` tasks in a Scala.js project require " +
+              "`fork in Test := false`.")
+        }
```

Comment:
Because it's a common dependency of all test task, and because from there on in the graph, things go wrong in sbt tasks. It seems fitting, but maybe there exists some more obscure task in sbt that would do the job better? If yes and unless you have a concrete suggestion now, that can always be improved later.